### PR TITLE
Pr90x l1t integration (v89.20) Mu isolation perf, EG new vars, Gen in L1TNtuple 

### PR DESCRIPTION
**Pull request by rekovic**
90x - l1t-integration from v89.15 to v89.20

Mu isolation performance improvements and extrapolation eta/phi to vtx. 
EG new vars (New EGamma var (HoverE) in  DataFormats/L1Trigger)
Gen fix in L1TNtuple 

Once approved will backport these commits onto #16966.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

File: L1Trigger/L1TNtuples/plugins/L1GenTreeProducer.cc [hunk 1]
```diff
@@ -92,10 +92,11 @@ L1GenTreeProducer::L1GenTreeProducer(const edm::ParameterSet& iConfig)
   genParticleToken_ = consumes<reco::GenParticleCollection>(iConfig.getUntrackedParameter<edm::InputTag>("genParticleToken"));
   pileupInfoToken_ = consumes<std::vector<PileupSummaryInfo> >(iConfig.getUntrackedParameter<edm::InputTag>("pileupInfoToken"));
   
+  l1GenData_ = new L1Analysis::L1AnalysisGeneratorDataFormat();
```
Comment (davidlange6) [inline L1Trigger/L1TNtuples/plugins/L1GenTreeProducer.cc]: this pointer needs to be deleted

Comment (VinInn) [inline L1Trigger/L1TNtuples/plugins/L1GenTreeProducer.cc]: please use std::unique_ptr and std::make_unique

Comment (rekovic) [inline L1Trigger/L1TNtuples/plugins/L1GenTreeProducer.cc]: OK. Will switch to smart ptrs.

**Comment by davidlange6** [comment]
even better, thanks

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**